src/os_common/filename.c

```diff
@@ -61,7 +61,7 @@ __wt_filename_construct(WT_SESSION_IMPL *session, const char *path, const char *
   uintmax_t id_1, uint32_t id_2, WT_ITEM *buf)
 {
     if (path != NULL && path[0] != '\0')
-        WT_RET(__wt_buf_catfmt(session, buf, "%s%s", path, __wt_path_separator()));
+        WT_RET(__wt_buf_fmt(session, buf, "%s%s", path, __wt_path_separator()));
```

Comment:
Can you explain this change and why it is necessary and maybe add a comment? Are there other callers of this function that depend on the append nature of `wt_buf_catfmt` here?

Reply:
I elaborated on that here: https://github.com/wiredtiger/wiredtiger/pull/11453#discussion_r1929099096 
and also here: https://github.com/wiredtiger/wiredtiger/pull/11453#discussion_r1924604103 I think the comment threads may have been prematurely resolved by Andrew. 

Essentially looping around buf_catfmt breaks things. No other caller depends on that behaviour from my investigation. I don't think adding a comment will make total sense either as it will be explaining behaviour that changed. I'll think about what I could say.

Reply:
5341e7207

Reply:
That reads slightly awkwardly to me. How about something like:
Don't concatenate to the buffer when a path is given. Callers providing a path may be in a loop and concatenation won't work in that case.

Reply:
5362f8a43